### DISCUSSION TITLE
Location of the original data source path

### DISCUSSION THREAD

**Discussion by dressito** [sepinf-inc/IPED#2059] [General]
Hello!
Is the path to the original data source accessible somewhere inside the case folder?
I can see that a `newDataSourceLocations.txt` is created in the \iped\data folder, but I have to manually point to the new location, after I open the case, for this file to be created.
I've created a batch script to create this file for a list of cases, but I have to supply the script with the path, that I get by first running a case, pointing to the new source location (mostly AD1 files), opening this txt file and copying the path.
So, I was wondering if my script can read this info somewhere in the case folder and do it automatically.

Thanks

**Comment by lfcnassif** [comment]
Hi @dressito. For dd, e01, ex01, vmdk, vhd and vhdx evidences, the original data source location can be found into the sleuth.db sqlite database, in the tsk_image_names table, since TSK library is used to decode those evidences.

For other evidence types, like folders, ad1 and ufdr, unfortunately the original evidence location is stored directly inside the iped case index, using the Lucene java library, and it cannot be easily read without using Lucene lib.

**Reply by dressito** [reply]
Thanks for the reply, @lfcnassif!
I will try to devise some other solution then...

One thing I'm curious... the user manual states that "IPED will try to find the evidence using different path combinations", but this look up "applies just to disk image evidences, not for UFDR or AD1 evidences".
But there are cases where IPED find the AD1 evidence, even though most of the time we do have to supply the new path in the dialog box. And, in both cases, the folders are always structured in the same way, but some (few) times the evidence is automatically found and usually is not.
Can you explain how the search for AD1 files is made by IPED?

**Reply by lfcnassif** [reply]
Sorry for late replying. Actually there is no search for AD1, the logic is here:
https://github.com/sepinf-inc/IPED/blob/183016b89418917cdae8532fcbaa3ef99167cd38/iped-engine/src/main/java/iped/engine/task/index/IndexItem.java#L976-L992

I think it works when the processed AD1 was into the same volume as the output case, so IPED derivates a relative path and finds it in any computer, if the AD1 was not moved. If it was located in another volume, a relative path can't be derived, so an absolute one is used. If the AD1 is not found in the absolute path, the dialog asking for the new image path pops up.

**Reply by dressito** [reply]
No worries!
Thanks for the info.... I will see what can I do to automate my workflow, based on this!
